### PR TITLE
fixes/balances-assets-transactions

### PR DESCRIPTION
**Pull request by leozera14**
Fixes for balances, assets and transactions

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -160,7 +160,7 @@
     "eslint-plugin-react-hooks": "^4.2.0",
     "eslint-plugin-typescript-sort-keys": "^1.5.0",
     "file-loader": "^6.1.0",
-    "fork-ts-checker-webpack-plugin": "6.5.2",
+    "fork-ts-checker-webpack-plugin": "^6.2.10",
```
Comment (lucaslovato) [inline package.json:163]: @claudiovb I think we don't need this, right?

File: source/pages/Home/Home.tsx [hunk 2]
```diff
@@ -119,9 +119,7 @@ export const Home = () => {
                 </p>
               </div>
 
-              <p id="fiat-ammount">
-                {isTestnet || isLoadingBalances ? null : fiatPriceValue}
-              </p>
+              <p id="fiat-ammount">{isTestnet ? null : fiatPriceValue}</p>
```
Comment (lucaslovato) [inline source/pages/Home/Home.tsx:122]: @lucasgabrielgsp will this need to be modified within your changes in the testnet PR?

File: source/pages/Home/Panel/ActivityPanel.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import React, { useCallback, useEffect, useState } from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
```
Comment (lucaslovato) [inline source/pages/Home/Panel/ActivityPanel.tsx:1]: these file changes are in another PR I don't know if we need to replicate them here.

File: source/pages/Home/Panel/AssetsPanel.tsx [hunk 4]
```diff
@@ -27,8 +27,11 @@ export const AssetsPanel = () => {
     assets.ethereum?.filter((token: any) => token.chainId === chainId)
       ?.length === 0;
 
+  const sysAssetsValidation =
+    assets.syscoin?.filter((asset) => asset.chainId === chainId)?.length === 0;
+
   const filterValidation = isBitcoinBased
-    ? assets.syscoin?.length === 0
+    ? assets.syscoin?.length === 0 || sysAssetsValidation
```
Comment (lucaslovato) [inline source/pages/Home/Panel/AssetsPanel.tsx:34]: this `||` can be inside `sysAssetsValidation`

File: source/pages/Home/Panel/components/Transactions/List.tsx [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import { uniqueId } from 'lodash';
-import React, { Fragment, useCallback } from 'react';
+import React, { Fragment, useCallback, useMemo } from 'react';
```
Comment (lucaslovato) [inline source/pages/Home/Panel/components/Transactions/List.tsx:2]: these file changes are in another PR I don't know if we need to replicate them here.
These changes make the PR harder to review

File: source/scripts/Background/controllers/MainController.ts [hunk 6]
```diff
@@ -648,24 +647,37 @@ const MainController = (walletState): IMainController => {
...
     assetsManager.utils
       .updateAssetsFromCurrentAccount(
         currentAccount,
         isBitcoinBased,
         activeNetwork.url,
+        activeNetwork.chainId,
         networks
       )
       .then((updatedAssets) => {
+        const validateIfIsInvalidDispatch =
```
Comment (lucaslovato) [inline source/scripts/Background/controllers/MainController.ts:659]: this is really hard to read, maybe more constants with the validations and a final const with the final validation can make this cleaner

File: source/scripts/Background/controllers/assets/syscoin.ts [hunk 7]
```diff
@@ -38,31 +40,38 @@ const SysAssetsControler = (): ISysAssetsController => {
...
       const isTokensAssetValid = tokensAsset && tokensAsset.length > 0;
 
       const validTokens = isTokensAssetValid ? tokensAsset : tokens;
-      console.log('validTokens', validTokens);
-      if (!validTokens || validTokens.length === 0) {
-        return [];
-      }
+
+      const preventUndefined =
+        typeof validTokens === 'undefined' || validTokens === undefined
```
Comment (lucaslovato) [inline source/scripts/Background/controllers/assets/syscoin.ts:51]: maybe lodash isNill or isEmpty can avoid these typeof comparisons with `undefined`

File: source/pages/Home/Home.tsx (hunk 2 above)
Comment (lucasgabrielgsp) [reply source/pages/Home/Home.tsx:122]: @lucaslovato no, it's not necessary. In testnet PR, I didn't change this structure. Is exactly the same

**Pull request merged**